src/fabric/src/fabric_view_row.erl

```diff
@@ -0,0 +1,351 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(fabric_view_row).
+-export([from_props/2]).
+-export([get_id/1, get_key/1, get_value/1, get_doc/1, get_worker/1]).
+-export([set_key/2, set_doc/2, set_worker/2]).
+-export([transform/1]).
+
+-include_lib("fabric/include/fabric.hrl").
+
+from_props(Props, Options) ->
+    case couch_util:get_value(view_row_map, Options, false) of
+        true ->
+            Row = lists:foldl(
+                fun({Key, Value}, ViewRow) -> ViewRow#{Key => Value} end,
+                #{},
+                Props
+            ),
```

Comment:
`maps:from_lis/2` should work here (`{view_row, maps:to_list(Props)}` ?)

Reply:
Yeah, you are right!